enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/APIKeyAuthenticator.java

```diff
@@ -61,27 +62,43 @@
 public class APIKeyAuthenticator extends APIKeyHandler {
 
     private static final Logger log = LogManager.getLogger(APIKeyAuthenticator.class);
+
+    private static String certAlias;
+    private static boolean apiKeySubValidationEnabled;
     private AbstractAPIMgtGatewayJWTGenerator jwtGenerator;
     private final boolean isGatewayTokenCacheEnabled;
-    private boolean apiKeySubValidationEnabled = false;
-
     private static final int IPV4_ADDRESS_BIT_LENGTH = 32;
     private static final int IPV6_ADDRESS_BIT_LENGTH = 128;
 
     public APIKeyAuthenticator() {
-        log.info("API key authenticator initialized.");
+        log.debug("API key authenticator initialized.");
         EnforcerConfig enforcerConfig = ConfigHolder.getInstance().getConfig();
         this.isGatewayTokenCacheEnabled = enforcerConfig.getCacheDto().isEnabled();
         if (enforcerConfig.getJwtConfigurationDto().isEnabled()) {
             this.jwtGenerator = BackendJwtUtils.getApiMgtGatewayJWTGenerator();
         }
-        Map<String, ExtendedTokenIssuerDto> tokenIssuers = ConfigHolder.getInstance().getConfig().getIssuersMap();
-        for (ExtendedTokenIssuerDto tokenIssuer : tokenIssuers.values()) {
-            if (APIConstants.KeyManager.APIM_PUBLISHER_ISSUER.equals(tokenIssuer.getName())) {
+        for (ExtendedTokenIssuerDto tokenIssuer : enforcerConfig.getIssuersMap().values()) {
+            if (APIConstants.KeyManager.APIM_APIKEY_ISSUER.equals(tokenIssuer.getName())) {
+                certAlias = tokenIssuer.getCertificateAlias();
                 apiKeySubValidationEnabled = tokenIssuer.isValidateSubscriptions();
                 break;
             }
         }
+
+        // For backward compatibility
```

Comment:
do we require this for new cc releases?

Reply:
Most of the case, we will be having the same use cases. So let's go with this for the moment. We can change these in the future if required.